haskell.nix/library/.github/workflows/check-nightly.yml

```diff
@@ -0,0 +1,39 @@
+on:
+  schedule:
+    # runs every Monday at 10:00 am UTC
+    - cron: '0 10 * * 1'
+
+jobs:
+  check-nightly:
+    # We use shared runners instead of our selfhosted ones because the stack integrates quite poorly with NixOS
+    runs-on: ubuntu-latest
+    outputs:
+      resolver: ${{ steps.nightly.outputs.resolver }}
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: build with nightly
+        id: nightly
+        run: |
+          # get the latest nightly resolver
+          resolver=$(curl -s -w %{redirect_url} https://www.stackage.org/nightly | sed 's:.*/::')
```

Comment:
Btw, you don't really need to do this `curl`-into-`sed` magic because just `nightly` is synonymous to the latest `nightly` available (`nightly-2024-08-13` at the moment of writing).

Reply:
Doesn't seem to work with `--resolver`:
```
λ stack build --resolver nigthly                                                                                                                                                                                                                             

Error: [S-775]
       Exception while reading snapshot from nigthly:
       
       YAML exception:
       Yaml file not found: /home/sereja/Code/work/nyan-interpolation/nigthly
```
```
λ stack --version                                                                                                                                                                                                                                            
3.1.1 x86_64 hpack-0.37.0
```

Reply:
There's a typo in your command, try `nightly`. I see that you also use the explicit resolver name in the error message, but I'm not sure if it's worth this complexity. The full name looks like `nightly-2024-09-12`, i. e. it contains the date, but usually you don't care and will test it with the latest available `nightly` anyway, even if it's newer than what was used on CI, because once somerthing breaks, it doesn't get fixed automatically in a future resolver.

Reply:
Oh, you're right, my bad. Pushed changes